tests/NuGet.Services.BasicSearchTests/V2SearchFunctionalTests.cs

```diff
@@ -37,7 +37,47 @@ public class V2SearchSkipFunctionalTests
             }
         }
 
-        [Fact(Skip = "The old V2 search service always includes prerelease versions, even if the parameter is set to false.")]
+        [Fact]
+        public async Task TreatsFirstIdTermAsExactMatch()
+        {
+            // Arrange
+            var packages = new[]
+            {
+                new PackageVersion("WebGrease", "1.6.0")
+            };
+
+            using (var app = await StartedWebApp.StartAsync(packages))
+            {
+                // Act
+                var partialIdResponse = await app.Client.GetAsync(new V2SearchBuilder
+                {
+                    Query = "version:1.6.0 id:Web",
+                    Prerelease = true,
+                }.RequestUri);
+                var partialId = await partialIdResponse.Content.ReadAsAsync<V2SearchResult>();
+
+                var wrongExactIdResponse = await app.Client.GetAsync(new V2SearchBuilder
+                {
+                    Query = "id:Web version:1.6.0",
```

Comment:
Why don't we support partial id matches here?

Reply:
Gallery expects this to work as exact match for V2 hijacking, e.g. `/api/v2/Packages(Id='Web',Version='1.6.0')`